### PR TITLE
Move to common_tests

### LINKED ISSUE #12: Move to common_test

### PR DESCRIPTION
**Pull request by hairyhum**
Fixes #12

### CODE REVIEW

File: test/src/rabbit_web_mqtt_SUITE.erl [hunk 1]
```diff
@@ -14,13 +14,58 @@
...
+    [
+      {group, non_parallel_tests}
+    ].
+
+groups() ->
+    [
+      {non_parallel_tests, [],
+       [connection_test
+        ,pubsub_test
+        ,disconnect_test
```
Comment (dumbbell) [inline test/src/rabbit_web_mqtt_SUITE.erl]: Please remove the `_test` suffix so they are not to be confused with EUnit tests.

**Comment by dumbbell** [comment]
Please move the test source files to `test` (instead of `test/src`).

Also, you should use a shorter name for the testsuite; eg. something like `system_SUITE` if the testsuite is doing system integration tests. There is no need to prefix the module with `rabbit_web_mqtt_` because there can't be namespace conflict. And it's easier to type `make ct-system` than `make ct-rabbit_web_mqtt`.

**Comment by michaelklishin** [comment]
Some of our test suites use "old" suite names, though. So I don't think it's a deal breaker.

File: Makefile [hunk 2]
```diff
@@ -18,8 +18,3 @@ ERLANG_MK_COMMIT = rabbitmq-tmp
 include rabbitmq-components.mk
 include erlang.mk
```
Comment (dumbbell) [inline Makefile:22]: You should add the following two lines between those two includes:

``` make
# FIXME: Remove rabbitmq_test as TEST_DEPS from here for now.
TEST_DEPS := $(filter-out rabbitmq_test,$(TEST_DEPS))
```

**Comment by dumbbell** [comment]
Not a deal breaker, but it makes it consistent with other converted projects.

**Comment by dumbbell** [comment]
You need to add `rabbit` to the `TEST_DEPS` before including `rabbitmq-components.mk`. GNU Make will complain about duplicated rules, but that's ok. Once all projects are moved, I will clean `rabbitmq-components.mk`.

Here is an example from the federation plugin:

``` make
PROJECT = rabbitmq_federation

DEPS = amqp_client
TEST_DEPS += rabbit
```

**Pull request merged**